Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from policyengine_us import Microsimulation
from policyengine_us.data.datasets import CPS
from policyengine_us.tools.dev.taxsim.generate_taxsim_tests import TaxSim35
import numpy as np
import pandas as pd

# Disable warnings
import warnings

warnings.filterwarnings("ignore")

STATES = ["MA", "MA", "NY", "WA"]
DISTANCE = 100

taxsim = TaxSim35()
sim = Microsimulation()

taxsim_df = taxsim.generate_from_microsimulation(
    CPS, 2022, None, True, False
).set_index("taxsim_taxsimid")


def get_federal_tax_against_taxsim_results():
    tax = sim.calc("income_tax")
    tax.index = sim.calc("tax_unit_id").values
    comparison_df = pd.DataFrame(index=sim.calc("tax_unit_id").values)
    comparison_df["policyengine_us"] = tax
    comparison_df["taxsim"] = taxsim_df.taxsim_fiitax
    relative_distance = np.absolute(
        comparison_df.policyengine_us - comparison_df.taxsim
    )
    return (relative_distance < DISTANCE).mean()


def get_state_income_tax_against_taxsim_results(state: str):
    in_state = sim.calc("tax_unit_state").values == state
    tax = sim.calc("state_income_tax")
    tax.index = sim.calc("tax_unit_id").values
    comparison_df = pd.DataFrame(
        dict(
            policyengine_us=tax,
            taxsim=taxsim_df.taxsim_siitax,
        ),
        index=sim.calc("tax_unit_id").values,
    )
    comparison_df = comparison_df[in_state]
    relative_distance = np.absolute(
        comparison_df.policyengine_us - comparison_df.taxsim
    )
    return (relative_distance < DISTANCE).mean()


results_df = pd.DataFrame(
    {
        "Region": ["Federal"] + STATES,
        "Percent close": [get_federal_tax_against_taxsim_results()]
        + [
            get_state_income_tax_against_taxsim_results(state)
            for state in STATES
        ],
    }
)
results_df["Percent close"] = results_df["Percent close"].apply(
    lambda x: f"{x:.1%}"
)
results_df.set_index("Region")
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mImportError[0m                               Traceback (most recent call last)
[0;32m/tmp/ipykernel_4061/2493350769.py[0m in [0;36m<cell line: 0>[0;34m()[0m
[1;32m      1[0m [0;32mfrom[0m [0mpolicyengine_us[0m [0;32mimport[0m [0mMicrosimulation[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 2[0;31m [0;32mfrom[0m [0mpolicyengine_us[0m[0;34m.[0m[0mdata[0m[0;34m.[0m[0mdatasets[0m [0;32mimport[0m [0mCPS[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      3[0m [0;32mfrom[0m [0mpolicyengine_us[0m[0;34m.[0m[0mtools[0m[0;34m.[0m[0mdev[0m[0;34m.[0m[0mtaxsim[0m[0;34m.[0m[0mgenerate_taxsim_tests[0m [0;32mimport[0m [0mTaxSim35[0m[0;34m[0m[0;34m[0m[0m
[1;32m      4[0m [0;32mimport[0m [0mnumpy[0m [0;32mas[0m [0mnp[0m[0;34m[0m[0;34m[0m[0m
[1;32m      5[0m [0;32mimport[0m [0mpandas[0m [0;32mas[0m [0mpd[0m[0;34m[0m[0;34m[0m[0m

[0;31mImportError[0m: cannot import name 'CPS' from 'policyengine_us.data.datasets' (/home/runner/work/policyengine-us/policyengine-us/policyengine_us/data/datasets/__init__.py)

